src/components/DocumentViewer/DocumentViewer.tsx

```diff
@@ -49,23 +56,40 @@ export const DocumentViewer: React.FunctionComponent<DocumentViewerProps> = ({ d
   const UndoRedactButton: React.FunctionComponent<ButtonProps> = ({ path }) => {
     return (
       <button
-        className="bg-red-500 hover:bg-red-700 text-white font-bold py-2 px-4 rounded"
+        className="bg-red-600 hover:bg-red-700 text-white font-bold py-2 px-4 rounded"
         onClick={() => toggleChoice(path)}
       >
-        Undo Redact
+        Undo
       </button>
     );
   };
   if (Object.keys(document).length) {
-    const data = flatten(document, "");
+    const data = flatten(getData(document), "");
     const sensitiveFields = findAllSensitiveFields(data);
     const hasSensitiveFields = sensitiveFields.length > 0;
 
+    if (hasSensitiveFields) {
+      const _redactionList: string[] = [...redactionList];
+      sensitiveFields.forEach((row: { path: string }) => {
+        _redactionList.push(row.path);
+      });
+      // setRedactionList(_redactionList);
```

Comment:
this somehow causes infinite re-rendering so I think it shouldn't be placed here, and instead should be placed at PrivacyFilter.tsx? (see other comment)

the purpose of this is: on file upload, I'll do one, and only one check for sensitive fields, then push them into the `redactionList`. checking it here seems like it causes an infinite loop...

Reply:
the reason is simple, when you will call `setRedactionList(_redactionList)`, a state change will be triggered, the component will be rendered but the call to `setRedactionList(_redactionList)` happens during rendering hence the infinite loop

Have a look to `useEffect`